sql/core/src/main/java/org/apache/spark/sql/execution/vectorized/WritableColumnVector.java

```diff
@@ -324,24 +315,27 @@ public void putDecimal(int rowId, Decimal value, int precision) {
   @Override
   public UTF8String getUTF8String(int rowId) {
     if (dictionary == null) {
-      ColumnarArray a = getByteArray(rowId);
-      return UTF8String.fromBytes(a.byteArray, a.byteArrayOffset, a.length);
+      return childColumns[0].getUTF8String0(getArrayOffset(rowId), getArrayLength(rowId));
```

Comment:
It looks risky if we do not make a copy. 

If we plan to avoid the unnecessary data copy by this API, we should rename the API name `getUTF8String` and check all the callers whether they do not break the assumption.

Reply:
This is a bit of a non-issue: the current on-heap code path already avoids making copies.

Reply:
It might be a bit better to use `arrayData()` instead of `childColumns[0]`, they are practically the same, but it makes the intent a bit clearer.